master/internal/proxy/proxy.go

```diff
@@ -51,55 +27,60 @@ type ProxyHTTPAuth func(echo.Context) (done bool, err error)
 
 // Proxy is an actor that proxies requests to registered services.
 type Proxy struct {
-	lock     sync.RWMutex
-	services map[string]*Service
-
 	HTTPAuth ProxyHTTPAuth
+	Services map[string]*Service
```

Comment:
these don't need to be exported

Reply:
In general, no. One of the main advantages of not exporting something is minimizing its API surface area so that people looking at the API can assume, if possible, that if something is exported, it's *intended* that it's used from outside. So, for example, users might assume they should interact with this map directly by adding to it, which would be a race condition.